feat: Improve Glossary entries for symm/asymm crypto & ciphers (enc algorithms) (#33396)
mdn/content: files/en-us/glossary/cipher/index.md

{{GlossarySidebar}}

> In {{glossary("cryptography")}}, a **cipher** is an algorithm that can {{glossary("encryption", "encrypt")}} {{glossary("Plaintext")}} to make it unreadable, and to {{glossary("decryption", "decrypt")}} the encoded data back to plaintext again.

Reviewer: should we use the lowercase one `plaintext` in the macro here? 🤔

Ciphers were common long before the information age (e.g., [substitution ciphers](https://en.wikipedia.org/wiki/Substitution_cipher), [transposition ciphers](https://en.wikipedia.org/wiki/Transposition_cipher), and [permutation ciphers](https://en.wikipedia.org/wiki/Permutation_cipher)), but none of them were cryptographically secure except for the [one-time pad](https://en.wikipedia.org/wiki/One-time_pad).

> [AES](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard), [RSA](<https://en.wikipedia.org/wiki/RSA_(cryptosystem)>) and [Blowfish](<https://en.wikipedia.org/wiki/Blowfish_(cipher)>) are examples of ciphers that are part of modern encryption standards and systems.

Reviewer: maybe we could add a linebreak between l11 & l12 here, and repharse it to make it more consistent like

```suggestion

In the modern age, ciphers have evloved dramatically. [AES](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard), [RSA](<https://en.wikipedia.org/wiki/RSA_(cryptosystem)>) and [Blowfish](<https://en.wikipedia.org/wiki/Blowfish_(cipher)>) are examples of ciphers that are integral part of contemporary encryption standards and systems.
```

Author: I'm fine with that one 👍🏻

Modern ciphers are designed to withstand attacks discovered through {{glossary("cryptanalysis")}}. There is no guarantee that all attack methods have been discovered, so each algorithm is [recommended for different purposes](/en-US/docs/Web/API/SubtleCrypto#supported_algorithms) based on known classes of attacks.

Ciphers operate either as [block ciphers](https://en.wikipedia.org/wiki/Block_cipher) on successive blocks (or buffers) of data, or as [stream ciphers](https://en.wikipedia.org/wiki/Stream_cipher) on a continuous data flow (often of sound or video).

> Ciphers also are classified according to how their {{glossary("key", "keys")}} are handled:

Reviewer: ```suggestion
Ciphers are also classified according to how their {{glossary("key", "keys")}} are handled:
```

- {{Glossary("Symmetric-key cryptography", "symmetric key")}} algorithms use the same key to encode and decode a message. The key also must be sent securely if the message is to stay confidential.

> - {{Glossary("Public-key cryptography", "asymmetric key")}} algorithms use one key for encryption and another for decryption.

Reviewer: ```suggestion
- {{Glossary("Public-key cryptography", "asymmetric key")}} algorithms use one key for encryption and the other for decryption.
```

## See also

- [Web Crypto API](/en-US/docs/Web/API/Web_Crypto_API)
- [SubtleCrypto supported algorithms](/en-US/docs/Web/API/SubtleCrypto#supported_algorithms)

> - [MDN Web Docs Glossary](/en-US/docs/Glossary)

Reviewer: wondering if this link could be remove, as it seems to be unnecessary ❓ 

(see discussion https://github.com/mdn/content/pull/33423#discussion_r1590843688)

```suggestion
- Glossary
```
